x-pack/filebeat/input/httpjson/chain.go

```diff
@@ -56,35 +56,31 @@
 
 package httpjson
 
-import (
-	"github.com/elastic/elastic-agent-libs/mapstr"
-	"github.com/elastic/elastic-agent-libs/transport/httpcommon"
-)
-
 // chainConfig for chain request.
 // Following contains basic call structure for each call after normal httpjson
 // call.
 type chainConfig struct {
-	Step stepConfig `config:"step" validate:"required"`
+	Step  *stepConfig  `config:"step,omitempty"`
+	While *whileConfig `config:"while,omitempty"`
 }
 
 // stepConfig will contain basic properties like, request.url,
 // request.method and replace parameter. Each step: request.url
 // will contain replace string with original URL to make a skeleton for the
 // call request.
 type stepConfig struct {
-	Request  requestChainConfig  `config:"request"`
+	Auth     *authConfig         `config:"auth"`
+	Request  requestConfig       `config:"request" validate:"required"`
 	Response responseChainConfig `config:"response,omitempty"`
 	Replace  string              `config:"replace,omitempty"`
 }
 
-type requestChainConfig struct {
-	URL        *urlConfig       `config:"url" validate:"required"`
-	Method     string           `config:"method" validate:"required"`
-	Body       *mapstr.M        `config:"body"`
-	Transforms transformsConfig `config:"transforms"`
-
-	Transport httpcommon.HTTPTransportSettings `config:",inline"`
+type whileConfig struct {
+	Auth     *authConfig         `config:"auth"`
+	Request  requestConfig       `config:"request" validate:"required"`
+	Response responseChainConfig `config:"response,omitempty"`
+	Replace  string              `config:"replace,omitempty"`
+	Till     *valueTpl           `config:"till"`
 }
```

Comment:
@efd6 Here since we are using an retryablehttp client to do the request .. by default if **request.retry.max_attempts** is not specified .. it will always perform 1 request and give up which keeps it turing incomplete.

Reply:
Yeah, OK. Can the documentation around the interaction between `till` and the `request.retry.max_attempts` be made clearer so that that is obvious; my naive reading was that it was the primary condition.

Reply:
@efd6 Sure , will update this in the document, it was a miss from my side . 👍

Reply:
Thanks, that looks good now.